sdk/nodejs/x/automation/events.ts

```diff
@@ -0,0 +1,198 @@
+    message: string;
+    color: string;
+    severity: "info" | "info#err" | "warning" | "error";
+    streamID?: number;
+    ephemeral?: boolean;
+}
+
+// PolicyEvent is emitted whenever there is a Policy violation.
+export interface PolicyEvent {
+    resourceUrn?: string;
+    message: string;
+    color: string;
+    policyName: string;
+    policyPackName: string;
+    policyPackVersion: string;
+    policyPackVersionTag: string;
+    enforcementLevel: "warning" | "mandatory";
+}
+
+// PreludeEvent is emitted at the start of an update.
+export interface PreludeEvent {
+    // config contains the keys and values for the update.
+    // Encrypted configuration values may be blinded.
+    config: Record<string, string>;
+}
+
+// SummaryEvent is emitted at the end of an update, with a summary of the changes made.
+export interface SummaryEvent {
+    // maybeCorrupt is set if one or more of the resources is in an invalid state.
+    maybeCorrupt: boolean;
+    // duration is the number of seconds the update was executing.
+    durationSeconds: number;
+    // resourceChanges contains the count for resource change by type. The keys are deploy.StepOp,
+    // which is not exported in this package.
+    resourceChanges: Record<string, number>;
+    // policyPacks run during update. Maps PolicyPackName -> version.
+    // Note: When this field was initially added, we forgot to add the JSON tag
+    // and are now locked into using PascalCase for this field to maintain backwards
+    // compatibility. For older clients this will map to the version, while for newer ones
+    // it will be the version tag prepended with "v".
+    policyPacks: Record<string, string>;
+}
+
+export enum DiffKind {
+    // add indicates that the property was added.
+    add = "add",
+    // addReplace indicates that the property was added and requires that the resource be replaced.
+    addReplace = "add-replace",
+    // delete indicates that the property was deleted.
+    delete = "delete",
+    // deleteReplace indicates that the property was deleted and requires that the resource be replaced.
+    deleteReplace = "delete-replace",
+    // update indicates that the property was updated.
+    update = "update",
+    // updateReplace indicates that the property was updated and requires that the resource be replaced.
+    updateReplace = "update-replace",
+}
+
+// PropertyDiff describes the difference between a single property's old and new values.
+export interface PropertyDiff {
+    // diffKind is the kind of difference.
+    diffKind: DiffKind;
+    // inputDiff is true if this is a difference between old and new inputs rather than old state and new inputs.
+    inputDiff: boolean;
+}
+
+// StepEventMetadata describes a "step" within the Pulumi engine, which is any concrete action
+// to migrate a set of cloud resources from one state to another.
+export interface StepEventMetadata {
+    // Op is the operation being performed, a deploy.StepOp.
+    op: string;
```

Comment:
This sorta goes for Go as well, but this field is an enum in the engine. It would be convenient to have enums in our SDKs as well unless there's a good reason not to throw a type on Op. 

Through my usage, I repeatedly had to reference the `deploy.StepOp` type in the engine. Might be a little more annoying for node developer (and anyone not familiar with the layout of pulumi/pulumi).